### PR TITLE
Feature/cieluv

### PR DESCRIPTION
**Pull request by masonium**
Implements the CIELUV color space, translating directly to and from `Xyz`.

First step towards `Hsluv` implementation.

### CODE REVIEW

File: palette/src/lch.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use rand::distributions::{Distribution, Standard};
 #[cfg(feature = "random")]
 use rand::Rng;
 
-use crate::color_difference::ColorDifference;
+use crate::{color_difference::ColorDifference};
```
Comment (Ogeon) [inline palette/src/lch.rs:11]: You can revert this change while fixing other parts.

File: palette/src/luv.rs [hunk 2]
```diff
@@ -0,0 +1,857 @@
...
+    Wp: WhitePoint,
+{
+    /// L\* is the lightness of the color. 0.0 gives absolute black and 100
+    /// give the brightest white.
+    pub l: T,
+
+    /// u* goes from -100 to 100.
+    pub u: T,
+
+    /// v* goes from -100 to 100.
```
Comment (Ogeon) [inline palette/src/luv.rs:54]: It's probably good to escape the `*` for good measure, to avoid accidental emphasis formatting.

File: palette/src/luv.rs [hunk 3]
```diff
@@ -0,0 +1,857 @@
...
+    }
+
+    fn clamp(&self) -> Luv<Wp, T> {
+	let mut c = *self;
+	c.clamp_self();
+	c
+    }
+
+    fn clamp_self(&mut self) {
+	self.l = clamp(self.l, T::zero(), from_f64(100.0));
```
Comment (Ogeon) [inline palette/src/luv.rs:306]: You could use `{min,max}_l()` here too.

File: palette/tests/hsluv_dataset/hsluv_dataset.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+use palette::convert::IntoColorUnclamped;
+use palette::white_point::D65;
+use palette::{Lch, Luv, Xyz};
+use std::collections::HashMap;
+
+#[derive(Clone, Debug)]
+struct HsluvExample {
+    name: String,
+    lch: Lch<D65, f64>,
```
Comment (Ogeon) [inline palette/tests/hsluv_dataset/hsluv_dataset.rs:16]: Is this the correct LCH? 😬

File: palette/src/luv.rs [hunk 5]
```diff
@@ -0,0 +1,857 @@
...
+	let prime_denom_recip = prime_denom.recip();
+	let prime_ref_denom_recip = (w.x + from_f64(15.0) * w.y + from_f64(3.0) * w.z).recip();
+
+	let u_prime: T = from_f64(4.0) * color.x * prime_denom_recip;
+	let u_ref_prime = from_f64(4.0) * w.x * prime_ref_denom_recip;
+	dbg!(u_ref_prime.to_f64().unwrap());
+
+	let v_prime: T = from_f64(9.0) * color.y * prime_denom_recip;
+	let v_ref_prime = from_f64(9.0) * w.y * prime_ref_denom_recip;
+	dbg!(v_ref_prime.to_f64().unwrap());
```
Comment (Ogeon) [inline palette/src/luv.rs:245]: These `dbg` lines should be removed.

**Review by Ogeon — COMMENTED** [review]
Nice! First step! I have only nit picks, really, except the `dbg` lines that got into the commit. They are also why the tests are failing. I also noticed the indentation was a bit mixed, so please run this through `rustfmt` as well. 🙂 Well done!

File: palette/tests/hsluv_dataset/hsluv_dataset.rs (hunk 4 above)
Comment (masonium) [reply palette/tests/hsluv_dataset/hsluv_dataset.rs:16]: I believe this is the appropriate LCH, but I'll remove it until I actually implement LCH in the next round so it's actually tested.

**Comment by masonium** [comment]
There are a few other files that `cargo fmt` modifies when running it over the whole repo (`palette/build/named.rs`, `palette/examples/color_scheme.rs`, `palette/src/named.rs`). I didn't include those with my formatting commit because they are technically unrelated, but I can just add them if you like.

File: palette/tests/hsluv_dataset/hsluv_dataset.rs (hunk 4 above)
Comment (masonium) [reply palette/tests/hsluv_dataset/hsluv_dataset.rs]: Silly me. That's *definitely* not the right `Lch`. I started working on `Lchuv` in a new branch on top of this one, and mentally I've already changed this. But of course it's not right in this branch yet 🤦🏾

Comment (Ogeon) [reply palette/tests/hsluv_dataset/hsluv_dataset.rs]: Parallel universes can be confusing like that. 😁

**Comment by Ogeon** [comment]
No need to include the additional files. I think this is ready for a merge. Thanks!

bors r+
